src/webview/macos/menu_bar.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright 2019-2021 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use cocoa::{
+  appkit::{NSApplication, NSEventModifierFlags, NSMenu, NSMenuItem},
+  base::{id, nil, selector},
+  foundation::{NSAutoreleasePool, NSProcessInfo, NSString},
+};
+use objc::{
+  declare::ClassDecl,
+  runtime::{Object, Sel},
+};
+use raw_window_handle::{HasRawWindowHandle, RawWindowHandle};
+use std::os::raw::c_void;
+use winit::window::Window;
+
+struct KeyEquivalent<'a> {
+  key: &'a str,
+  masks: Option<NSEventModifierFlags>,
+}
+
+// We do not support custom menu right now as we should fix our focus
+// first as it prevent a good use of the menu bar
+// Related to https://github.com/tauri-apps/wry/issues/184
+extern "C" fn handle_menu(_this: &Object, _cmd: Sel, _item: id) {
+  // use `handleCustomMenu` as selector to trigger this function
+  println!("Custom menu hanlder");
+}
+
+// Safety: objc runtime calls are unsafe
+pub(crate) unsafe fn add_menu_methods(decl: &mut ClassDecl) {
+  decl.add_method(
+    selector("handleCustomMenu:"),
+    handle_menu as extern "C" fn(&Object, _cmd: Sel, item: id),
+  );
+  decl.add_ivar::<*mut c_void>("eventProxy");
+}
+
+// Safety: objc runtime calls are unsafe
+pub(crate) unsafe fn create_menu(window: &Window) {
+  // make sure our raw window is available
+  if let RawWindowHandle::MacOS(_window) = window.raw_window_handle() {
```

Comment:
Could we check it with the method from `WindowExtMacOS` trait? Like checking if it's not null.

Reply:
Thanks I'll check!

Reply:
So with my tests, we don't even require to check `RawWindowHandle::MacOS(_window) = window.raw_window_handle()`, I switched to the `sharedApplication` method and worked fine and forgot to test without the raw_window_handle

It may be required for the custom implementation of the menu handler.

Or we could also check to do something like you've done with file_drop with callback Fn but I think the custom menu will be another scope. Because I don't want to register a method for each menu item as it makes no sense.

Reply:
yeah custom menu should be another scope. Let's leave it to future us to worry.